lib/bakeAmbientOcclusion.js

```diff
@@ -43,20 +48,323 @@ function bakeAmbientOcclusion(gltf, options) {
     // Generate triangle soup
     var raytracerScene = generateRaytracerScene(gltf, scene, options);
 
-    // Raytrace to a new texture for each primitive
+    // Raytrace for each primitive and add to the gltf
     var parameters = {
         raytracerScene: raytracerScene,
         resolution: options.resolution
     };
-    NodeHelpers.forEachPrimitiveInScene(gltf, scene, raytracePrimitiveTexels, parameters);
 
-    // Add to the gltf
-    bakeToDiffuse(gltf, scene, options.resolution, raytracerScene);
+    if (options.toVertex) {
+        NodeHelpers.forEachPrimitiveInScene(gltf, scene, raytraceAtTriangleCenters, parameters);
+        if (options.triangleAverage) {
+            NodeHelpers.forEachPrimitiveInScene(gltf, scene, raytraceOverTriangleSamples, parameters);
+        }
+
+        bakeToVertices(gltf, scene, raytracerScene, options);
+    } else {
+        NodeHelpers.forEachPrimitiveInScene(gltf, scene, raytraceToTexels, parameters);
+        bakeToTexture(gltf, scene, options.resolution, raytracerScene);
+    }
 }
 
-////////// adding to the gltf //////////
+////////// adding to the gltf by vertex //////////
+
+// helper for bakeToVertices
+function checkShadingChain(primitive, meshPrimitiveID, parameters) {
+    if (parameters.aoBufferByPrimitive.hasOwnProperty(meshPrimitiveID)) {
+        var materialID = primitive.material;
+        var techniqueID = parameters.materials[materialID].technique;
+        var programID = parameters.techniques[techniqueID].program;
+        var program = parameters.programs[programID];
+        var fragmentShaderID = program.fragmentShader;
+        var vertexShaderID = program.vertexShader;
+
+        parameters.materialCloneIDs[materialID] = '';
+        parameters.techniqueCloneIDs[techniqueID] = '';
+        parameters.programCloneIDs[programID] = '';
+        parameters.vertexShaderCloneIDs[vertexShaderID] = '';
+        parameters.fragmentShaderCloneIDs[fragmentShaderID] = '';
+    }
+}
+
+function cloneJsonAsNeeded(id, cloneIds, items) {
+    var cloneID = cloneIds[id];
+    if (defined(cloneID)) {
+        if (cloneID === '') { // No clone exists yet. Make one!
+            cloneID = id + "_noAO";
+            items[cloneID] = clone(items[id]);
+            cloneIds[id] = cloneID;
+        }
+        return cloneID;
+    } else {
```

Comment:
Drop the `else`.

Just `return id;`

See https://github.com/AnalyticalGraphicsInc/cesium/tree/master/Documentation/Contributors/CodingGuide#functions